test/e2e/spec/test.spec.js

```diff
@@ -0,0 +1,175 @@
+    const page = await browser.newPage();
+    const assetTitle = `test@${rHash}`;
+    await page.goto(`${url}/assets/title/${assetTitle}`, {waitUntil: 'networkidle'});
+
+    // Checking asset title
+    const pageTitle = await page.title();
+    expect(pageTitle).to.equal(assetTitle);
+  }));
+
+  it('goes to the embed iframe',  test(async (browser, _, context) => {
+    const page = browser.currentPage;
+    const embedStream = await getEmbedStream(page.mainFrame());
+
+    // Set the embedStream
+    const embedStreamUrl = embedStream.url();
+    context.setData({embedStreamUrl});
+    
+    // TODO: (bc) CHANGE WHEN THIS ISSUE IS SOLVED: https://github.com/GoogleChrome/puppeteer/issues/684
+
+    // Going to the embed stream url
+    await page.goto(embedStreamUrl, {waitUntil: 'networkidle'});
+    
+    // The URLs should match
+    expect(page.url()).to.equal(embedStreamUrl);
+  }));
+
+  it('not logged in user clicks my profile tab', test(async (browser, {$}) => {
+    const page = browser.currentPage;
+
+    // Clicking profile tab
+    const profileTab = await page.$($.Stream.myProfileTab);
+    expect(profileTab).to.not.equal(null);
+    await profileTab.click();
+
+    await page.waitForSelector($.Stream.tabContent);
+
+    // Not signed in users should get not-logged-in messages
+    const notLoggedInMessage = await page.waitForSelector($.Stream.notLoggedInMessage);
+    expect(notLoggedInMessage).to.not.equal(null);
+  }));
+
+  it('creates an user and user logs in', test(async (browser, {opts: {url, typeDelay}, $, data}) => {
+    const page = await browser.currentPage;
+    const rHash = murmur3(uuid.v4());
+    const formData = {
+      email: `test_${rHash}@test.test`,
+      username: `test${rHash}`,
+      password: `testpassword${rHash}`
+    };
+
+    // Go to Login View
+    await page.goto(`${url}/embed/stream/login`, {waitUntil: 'networkidle'});
+    
+    // Click Register
+    await page.click($.Login.register);
+    await page.focus($.Login.emailField);
+
+    // Filling the form
+    await page.type(formData.email, {delay: typeDelay});
+    await page.focus($.Login.usernameField);
+    await page.type(formData.username, {delay: typeDelay});
+    await page.focus($.Login.passwordField);
+    await page.type(formData.password, {delay: typeDelay});
+    await page.focus($.Login.confirmPasswordField);
+    await page.type(formData.password, {delay: typeDelay});
+
+    // Clicking Sign-Up
+    await page.click($.Login.signUpButton);
+
+    // Clicking Login
+    await page.waitForSelector($.Login.loginButton);
+    await page.click($.Login.loginButton);
+    await page.waitForNavigation({waitUntil: 'networkidle'});
+
+    // Going to the Embed Stream
+    await page.goto(data.embedStreamUrl, {waitUntil: 'networkidle', networkIdleTimeout: 3000});
+
+    // Checking logged-in Username
+    const userBoxEl = await page.waitForSelector($.Stream.authUserboxUsername);
+    expect(userBoxEl).to.not.equal(null);
```

Comment:
We could add a tiny bit more to this test to check to see if the username was what we expected.